### PR TITLE
Allow to override the server url with $OWNCLOUD_OVERRIDE_SERVER_URL

### PR DESCRIPTION
**Pull request by TheOneRing**
Depends on: https://github.com/owncloud/client/pull/8771

### CODE REVIEW

**Comment by TheOneRing** [comment]
@SamuAlfageme :)

File: src/libsync/theme.h [hunk 1]
```diff
@@ -160,9 +160,16 @@ class OWNCLOUDSYNC_EXPORT Theme : public QObject
      * Setting a value here will pre-define the server url.
      *
      * The respective UI controls will be disabled
+     * Deprecated: Use overrideServerUrlV2 as it allows overrides
      */
+    Q_DECL_DEPRECATED_X("Use overrideServerUrlV2")
     virtual QString overrideServerUrl() const;
```
Comment (fmoc) [inline src/libsync/theme.h:166]: What's the incentive to keep this function rather than changing it directly?

File: src/libsync/theme.cpp [hunk 2]
```diff
@@ -318,6 +319,17 @@ QString Theme::overrideServerUrl() const
     return QString();
 }
 
+QString Theme::overrideServerUrlV2() const
+{
+    static auto serverOverride = qEnvironmentVariable("OWNCLOUD_OVERRIDE_SERVER_URL");
```
Comment (fmoc) [inline src/libsync/theme.cpp:324]: Can be `const`.

File: src/libsync/theme.h (hunk 1 above)
Comment (TheOneRing) [reply src/libsync/theme.h:166]: The reason is that the auto generated themes define this function...

Comment (fmoc) [reply src/libsync/theme.h:166]: I see.

I suppose the `V2` suffix is the reason for this confusion. The name should give an idea what the method does. This one is just a shallow wrapper that allows overrides via an environment variable, so I guess that fact should somehow end up in the method name.

**Pull request merged**